### ISSUE TITLE
prometheus_remote_write output out of order labels

### ISSUE THREAD

**Issue by Aaron-ML** [fluent/fluent-bit#6972]
## Bug Report

**Describe the bug**

It looks like the prometheus_remote_write output has issues with out of order labels, and is further exacerbated by newer versions of Thanos receiver blocking out of spec labels that do not conform with the remote write spec: https://github.com/thanos-io/thanos/pull/5508

**To Reproduce**
- Setup thanos receiver to accept remote-write samples
- Setup output to ship metrics via prometheus_remote_write endpoint:

- Steps to reproduce the problem:

**Expected behavior**

Metrics shipped should have labels in order following the standard referenced [here](https://docs.google.com/document/d/1LPhVRSFkGNSuU1fBd81ulhsCPR4hkSZyyBj1SZ8fWOM/edit#)
```
- MUST NOT contain repeated label names.
- MUST have label names sorted in lexicographical order.
- MUST NOT contain any empty label names or values.
``` 

**Your Environment**

* Version used: Fluentbit 2.0
* Configuration:
```
[OUTPUT] # Send metrics data to the remote metrics sink
    Name                 prometheus_remote_write
    Match                metrics.*
    Host                 ${METRICS_HOST}
    Port                 443
    URI                  ${METRICS_URI}
    Proxy                ${HTTPS_PROXY}
    Log_response_payload True
    Tls                  On
    Tls.verify           On
    tls.ca_file          ${SSL_CERT_FILE}
    # Add some labels
    add_label            host ${HOST}
    add_label            customer_tenant_id ${TENANT_ID}
    add_label            version ${_VERSION}
```
Log output:
```
level=debug ts=2023-03-02T00:13:06.033063041Z caller=writer.go:85 component=receive component=receive-writer tenant=default-tenant msg="Out of order labels in the label set" lset="labels:<name:\"__name__\" value:\"nginx_up\" > labels:<name:\"host\" value:\"redacted\" > labels:<name:\"customer_tenant_id\" value:\"redacted\" > labels:<name:\"version\" value:\"v1.2.0-b247\" > "
```
* Operating System and version: Debian/Kubernetes
* Filters and plugins: prometheus_remote_write

**Additional context**

This is currently impacting our ability to send metrics from a variety of hosts that are deployed across the country.

**Comment by leonardo-albertovich** [comment]
As the person who wrote both the encoder and output plugin I can confirm that it does not enforce this ordering policy but I am not entirely sure of how hard it would be to do it or how long it will take, I will think about it and try to assess it but I don't expect to be able to address this immediately.

Sorry for the annoyance, don't hesitate to ping me if I forget about this.

**Comment by bryandicarlo** [comment]
@leonardo-albertovich any luck?  Thank you!

**Comment by leonardo-albertovich** [comment]
Hi @bryandicarlo, sorry but I haven't been able to put any time on this. I have an idea on how to approach this but it's not a trivial thing and I don't want to rush it and cause any issues.

**Comment by Tiduster** [comment]
@leonardo-albertovich : thanks a lot for this plugin.
We used it for 2 years in production on hundred of hardware controllers.

We just upgraded our Thanos to latest version and are also facing this issue.
No metric is written to Thanos because labels are not in alphabetical order.

I understand perfectly if you don't have the time to fix this issue.
Should we consider this plugin deprecated and move to another solution?

Best regards,
